src/MoBi.Presentation/Tasks/Interaction/BaseInitialConditionsTask.cs

```diff
@@ -0,0 +1,228 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using MoBi.Assets;
+using MoBi.Core.Commands;
+using MoBi.Core.Domain.Builder;
+using MoBi.Core.Domain.Extensions;
+using MoBi.Core.Domain.Model;
+using MoBi.Core.Domain.Services;
+using MoBi.Core.Services;
+using MoBi.Presentation.DTO;
+using MoBi.Presentation.Mappers;
+using MoBi.Presentation.Tasks.Edit;
+using OSPSuite.Assets;
+using OSPSuite.Core.Commands.Core;
+using OSPSuite.Core.Domain;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Core.Domain.Formulas;
+using OSPSuite.Core.Domain.Services;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+
+namespace MoBi.Presentation.Tasks.Interaction
+{
+   public abstract class BaseInitialConditionsTask<TBuildingBlock> : StartValuesTask<TBuildingBlock, InitialCondition> where TBuildingBlock : class, IBuildingBlock<InitialCondition>
```

Comment:
A bit confused with this base task .We already have base presenter . Canot the concreted presenter instantiate the class with the right generics? It feels like we are duplicating the hierarchy twice